pubsub/cloud-client/subscriber.py

```diff
@@ -171,6 +221,113 @@ def update_subscription(project_id, subscription_name, endpoint):
+    # TODO subscription_name = "Your Pub/Sub subscription name"
+    # This is an existing dead letter topic that the subscription with dead
+    # letter policy will forward dead letter messages to.
+    # TODO dead_letter_topic_name = "Your Pub/Sub dead letter topic name"
+
+    subscriber = pubsub_v1.SubscriberClient()
+    topic_path = subscriber.topic_path(project_id, topic_name)
+    subscription_path = subscriber.subscription_path(
+        project_id, subscription_name
+    )
+    dead_letter_topic_path = subscriber.topic_path(
+        project_id, dead_letter_topic_name
+    )
+
+    subscription_before_update = subscriber.get_subscription(subscription_path)
+    print("Before the update: {}".format(subscription_before_update))
+
+    subscription_before_update.HasField("dead_letter_policy")
+
+    assert 'dead_letter_policy' in properties
+    update_mask = {"paths": {"dead_letter_policy": {"max_delivery_attempts"}}}
+
+    # Construct a dead letter policy you expect to have after the update.
+    dead_letter_policy = DeadLetterPolicy(
+        dead_letter_topic=dead_letter_topic_path, max_delivery_attempts=20
+    )
+
+    # Construct the subscription with the dead letter policy you expect to have
+    # after the update. Here, values in the required fields (name, topic) help
+    # identify the subscription.
+    subscription = pubsub_v1.types.Subscription(
+        name=subscription_path,
+        topic=topic_path,
+        dead_letter_policy=dead_letter_policy,
+    )
+
+    subscription_after_update = subscriber.update_subscription(
+        subscription, update_mask
+    )
+
+    print("After the update: {}".format(subscription_after_update))
+
+    subscriber.close()
+    # [END pubsub_dead_letter_update_subscription]
+    return subscription_after_update
+
+
+def remove_dead_letter_policy(project_id, topic_name, subscription_name):
+    """Remove dead letter policy from a subscription."""
+    # [START pubsub_dead_letter_remove]
+    from google.cloud import pubsub_v1
+    from google.cloud.pubsub_v1.types import DeadLetterPolicy
+
+    # TODO project_id = "Your Google Cloud Project ID"
+    # This is an existing topic that the subscription with dead letter policy
+    # is attached to.
+    # TODO topic_name = "Your Pub/Sub topic name"
+    # This is an existing subscription with a dead letter policy.
+    # TODO subscription_name = "Your Pub/Sub subscription name"
+
+    subscriber = pubsub_v1.SubscriberClient()
+    topic_path = subscriber.topic_path(project_id, topic_name)
+    subscription_path = subscriber.subscription_path(
+        project_id, subscription_name
+    )
+
+    subscription_before_update = subscriber.get_subscription(subscription_path)
+    print("Before removing the policy: {}".format(subscription_before_update))
+
+    update_mask = {
+        "paths": {"dead_letter_policy": {"dead_letter_topic"}},
+        "paths": {"dead_letter_policy": {"max_delivery_attempts"}},
+    }
```

Comment:
Can you add a comment explaining what this does?

Reply:
Done.